PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -413,7 +432,7 @@ def produceMET(self, process,  metType, metModuleSequence, postfix):
             process.load("PhysicsTools.PatUtils.patPFMETCorrections_cff")
             
         if postfix != "" and metType == "PF" and not hasattr(process, 'pat'+metType+'Met'+postfix):
-            noClonesTmp = [ "particleFlowDisplacedVertex", "pfCandidateToVertexAssociation" ]
+            noClonesTmp = [ "particleFlowDisplacedVertex", "pfCandidateToVertexAssociation" ] if not self._parameters["Puppi"].value else []
```

Comment:
On 9/5/16 2:37 AM, mariadalfonso wrote:
> In
> PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py
> <https://github.com/cms-sw/cmssw/pull/15565#discussion_r77495031>:
> 
>> @@ -413,7 +432,7 @@ def produceMET(self, process,  metType, metModuleSequence, postfix):
>>              process.load("PhysicsTools.PatUtils.patPFMETCorrections_cff")
>>              
>>          if postfix != "" and metType == "PF" and not hasattr(process, 'pat'+metType+'Met'+postfix):
>> -            noClonesTmp = [ "particleFlowDisplacedVertex", "pfCandidateToVertexAssociation" ]
>> +            noClonesTmp = [ "particleFlowDisplacedVertex", "pfCandidateToVertexAssociation" ] if not self._parameters["Puppi"].value else []
> 


> @slava77 <https://github.com/slava77>
> For puppi this track association I think is also mis-leading as this one
> association is a bit different than the one used inside the charge
> hadrons. So prefer to have this disabled


I looked at the version before your last updates and it was the same as
the standard PF.
Please remove.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15565/files/4aa761b6de91fa35c59cd9da7595ba43099d2e68#r77495031>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbixW6YD9vUmaSo9VBjL90wPfbvZcks5qm-LIgaJpZM4Jq5Yu>.
>